Doc fixes

bcoca opened the pull request:
##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->

 - Docs Pull Request

##### COMPONENT NAME
<!--- Name of the module/plugin/module/task -->
docs templating

##### ANSIBLE VERSION
<!--- Paste verbatim output from “ansible --version” between quotes below -->
```

```

##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->

bcoca committed:
cleanup to reflect current builds

bcoca committed:
consolidate templating docs and minor rewording

bcoca committed:
new templating intro page

alikins commented:
Seeing some warnings on playbooks_templating.rst

```
/home/adrian/src/ansible/docs/docsite/rst/playbooks_templating.rst:9: ERROR: Content block expected for the "seealso" directive; none found.
/home/adrian/src/ansible/docs/docsite/rst/playbooks_templating.rst:12: WARNING: toctree contains reference to nonexisting document u':doc:`playbooks`'
/home/adrian/src/ansible/docs/docsite/rst/playbooks_templating.rst:12: WARNING: toctree contains reference to nonexisting document u'    An introduction to playbooks'
/home/adrian/src/ansible/docs/docsite/rst/playbooks_templating.rst:12: WARNING: toctree contains reference to nonexisting document u':doc:`playbooks_conditionals`'
/home/adrian/src/ansible/docs/docsite/rst/playbooks_templating.rst:12: WARNING: toctree contains reference to nonexisting document u'    Conditional statements in playbooks'
/home/adrian/src/ansible/docs/docsite/rst/playbooks_templating.rst:12: WARNING: toctree contains reference to nonexisting document u':doc:`playbooks_loops`'
/home/adrian/src/ansible/docs/docsite/rst/playbooks_templating.rst:12: WARNING: toctree contains reference to nonexisting document u'    Looping in playbooks'
/home/adrian/src/ansible/docs/docsite/rst/playbooks_templating.rst:12: WARNING: toctree contains reference to nonexisting document u':doc:`playbooks_roles`'
/home/adrian/src/ansible/docs/docsite/rst/playbooks_templating.rst:12: WARNING: toctree contains reference to nonexisting document u'    Playbook organization by roles'
/home/adrian/src/ansible/docs/docsite/rst/playbooks_templating.rst:12: WARNING: toctree contains reference to nonexisting document u':doc:`playbooks_best_practices`'
/home/adrian/src/ansible/docs/docsite/rst/playbooks_templating.rst:12: WARNING: toctree contains reference to nonexisting document u'    Best practices in playbooks'
/home/adrian/src/ansible/docs/docsite/rst/playbooks_templating.rst:12: WARNING: toctree contains reference to nonexisting document u'    Have a question?  Stop by the google group!'
/home/adrian/src/ansible/docs/docsite/rst/playbooks_templating.rst:12: WARNING: toctree contains reference to nonexisting document u'    #ansible IRC chat channel'
```

And playbooks_filters_ipaddr.rst isnt in an toc...
```
/home/adrian/src/ansible/docs/docsite/rst/playbooks_filters_ipaddr.rst:: WARNING: document isn't included in any toctree
```

bcoca committed:
fixed warnings as per feedback

bcoca commented:
fixed the seealso and toctree for playbooks_template, filters_ipadr is not in toctree on purpose, it is linked from playbooks_filters, added `:orphan:` to remove that warning.

dharmabumstead committed:
Update playbooks_filters.rst

Edited for clarity.

dharmabumstead committed:
Update playbooks_templating.rst

Light edits

dharmabumstead reviewed (approved):
Approved with edits. Thanks.

dharmabumstead commented:
Thanks @bcoca!

bcoca commented on docs/docsite/rst/playbooks_templating.rst:
```diff
@@ -2,9 +2,9 @@ Templating (Jinja2)
 
 As already referenced in the variables section, Ansible uses Jinja2 templating to enable dynamic expressions and access to variables.
-Not only that, it greatly expands the number of filters and tests available as well as adding it's own new plugin type, lookups.
+Jinja2 also greatly expands the number of filters and tests available, as well as adding a new plugin type: lookups.
```
Ansible .. not Jinja2
